core/src/chain/entry.rs

```diff
@@ -42,10 +33,14 @@ impl Entry {
     }
 
     /// hashes the entry
-    pub fn hash(&self) -> u64 {
-        let mut hasher = DefaultHasher::new();
-        Hash::hash(&self, &mut hasher);
-        hasher.finish()
+    pub fn hash(&self) -> String {
+        // @TODO - this is the wrong string being hashed
+        // @see https://github.com/holochain/holochain-rust/issues/103
+        let string_to_hash = self.content.clone();
+
+        // @TODO the hashing algo should not be hardcoded
+        // @see https://github.com/holochain/holochain-rust/issues/104
```

Comment:
Yes, I think we should define a trait perhaps called `HolochainHash`, and be returning that here. There's also the issue of the NullHash, i.e. you just switched from `0` to `""` and that should also be an abstraction, but lets keep that in #104